lib/ansible/modules/packaging/language/pip.py

```diff
@@ -417,12 +474,67 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err
 
 
+class Package:
+    """Python distribution package metadata wrapper.
+
+    A wrapper class for Requirement, which provides
+    API to parse package name, version specifier,
+    test whether a package is already satisfied.
+    """
+
+    def __init__(self, name_string, version_string=None):
+        self._plain_package = False
+        self.package_name = name_string
+        self._requirement = None
+        if name_string.startswith('file:') or _is_vcs_url(name_string):
+            return
+
+        self._plain_package = True
+        if version_string:
+            version_string = version_string.lstrip()
+            separator = '==' if version_string[0].isdigit() else ' '
+            name_string = separator.join((name_string, version_string))
+        try:
+            self._requirement = Requirement.parse(name_string)
+        except ValueError as e:
+            raise ValueError("Can not parse package name '%s', error: '%s'" % (name_string, to_native(e)))
```

Comment:
`six.raise_from(ValueError("Can not parse package name '%s', error: '%s'" % (name_string, to_native(e))), e)`

Reply:
Is this still necessary? There should be test coverage for this line by adding some broken requirements specification.